### PR TITLE
Changes to fix TheiaSfm dependencies

### PR DESCRIPTION
**Pull request by KindDragon**

### CODE REVIEW

**Comment by KindDragon** [comment]
@ras0219-msft TheiaSfm library can be only compiled with MSVC as static library, but they using some internal libraries that can only be compiled as shared library. Because of that vcpkg produce warning:
```
DLLs should not be present in a static build, but the following DLLs were found:

    d:/Work/vcpkg/packages/theia_x64-windows/debug/bin/vlfeat-debug.dll
    d:/Work/vcpkg/packages/theia_x64-windows/bin/flann_cpp.dll
    d:/Work/vcpkg/packages/theia_x64-windows/bin/vlfeat.dll
    d:/Work/vcpkg/packages/theia_x64-windows/debug/bin/flann_cpp-debug.dll
```
How disable this warning?

File: ports/openimageio/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: openimageio
+Version: 1.7.15
+Build-Depends: openexr
```
Comment (codicodi) [inline ports/openimageio/CONTROL]: I think oiio should have much more dependencies:
https://github.com/OpenImageIO/oiio/blob/master/CMakeLists.txt#L63-L99

Comment (codicodi) [inline ports/openimageio/CONTROL]: Currently, on a clean vcpkg instance build fails with
```
CMake Error at C:/Program Files/CMake/share/cmake-3.8/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find TIFF (missing: TIFF_LIBRARY TIFF_INCLUDE_DIR)
```

Comment (KindDragon) [inline ports/openimageio/CONTROL]: OIIO just ignore dependencies that it can't find, except openexr

Comment (codicodi) [inline ports/openimageio/CONTROL]: You didn't gen a error about missing tiff? Anyway, I think all dependencies should be explicitly disabled or enabled for reproducibility reasons

Comment (KindDragon) [inline ports/openimageio/CONTROL]: I don't want plug in all supported dependencies like Qt

Comment (codicodi) [inline ports/openimageio/CONTROL]: Then disable Qt. AFAIK it is not needed by the library anyway

**Comment by ras0219-msft** [comment]
@KindDragon For now, the best way would be to always claim it's a dynamic build.
```cmake
set(VCPKG_LIBRARY_LINKAGE dynamic)
```

**Comment by KindDragon** [comment]
@ras0219-msft but Theia library cannot be built as a dynamic library (cannot export all symbols).

**Comment by ras0219-msft** [comment]
Yes, it's somewhat misleading but we currently interpret "static" to mean "no DLLs" and "dynamic" to mean "some DLLs". Therefore, "dynamic" is currently the best categorization because there are "some DLLs".

**Comment by KindDragon** [comment]
@ras0219-msft I mean that when I tried with `VCPKG_LIBRARY_LINKAGE=dynamic` and with CMake option `BUILD_SHARED_LIBS=OFF`, CMake tried to build Theia as dll :(

**Pull request closed without merging**